src/jwks.cc

```diff
@@ -413,6 +484,34 @@ void Jwks::createFromPemCore(const std::string& pkey_pem) {
       key_ptr->ec_key_.reset(EVP_PKEY_get1_EC_KEY(evp_pkey.get()));
       key_ptr->kty_ = "EC";
       break;
+    case EVP_PKEY_ED25519: {
+      uint8_t raw_key[ED25519_PUBLIC_KEY_LEN];
+      size_t out_len = ED25519_PUBLIC_KEY_LEN;
+      if (EVP_PKEY_get_raw_public_key(evp_pkey.get(), raw_key, &out_len) != 1 ||
+          out_len != ED25519_PUBLIC_KEY_LEN) {
+        updateStatus(Status::JwksPemGetRawEd25519Error);
+        return;
+      }
+      key_ptr->okp_key_raw_ =
+          std::string(reinterpret_cast<const char*>(raw_key), out_len);
```

Comment:
can we move line 488 to 495 into a function so that code can be shared by X25519 case?

Reply:
This seems no longer necessary with X25519 removed, but if you think it's too much code in a case statement let me know and I'll make a function.

Reply:
With X25519 removed.  this is fine